### PR TITLE
libgtop: update to 2.40.0

### PR DESCRIPTION
**Pull request by barracuda156**
Closes: https://trac.macports.org/ticket/63976

#### Description

Update, also fixes the build

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by barracuda156** [comment]
```
  main.c:226:35: error: incompatible pointer to integer conversion passing 'char[1024]' to parameter of type 'pid_t' (aka 'int') [-Wint-conversion]
                  (server, &resp->u.data.proc_io, parameter);
                                                  ^~~~~~~~~
  ../../include/glibtop/procio.h:78:74: note: passing argument to parameter 'pid' here
  void glibtop_get_proc_io_l (glibtop *server, glibtop_proc_io *buf, pid_t pid);
                                                                           ^
  main.c:244:8: warning: format specifies type 'unsigned long long' but the argument has type 'guint64' (aka 'unsigned long') [-Wformat]
                              cmnd->command);
                              ^~~~~~~~~~~~~
  10 warnings and 1 error generated.
```

I will add a fix for this.

File: gnome/libgtop/Portfile [hunk 1]
```diff
@@ -4,36 +4,40 @@ PortSystem          1.0
...
+                    path:lib/pkgconfig/glib-2.0.pc:glib2 \
+                    port:xorg-libXau
 
-patchfiles          patch-libgtop-sysdeps.m4.diff \
-                    patch-add-null-pointer-checker.diff
+patchfiles          patch-libgtop-sysdeps.m4.diff
 
-gobject_introspection yes
+gobject_introspection \
+                    yes
```
Comment (pmetzger) [inline gnome/libgtop/Portfile:40]: I'm not sure splitting this into two lines makes things more readable but I don't care that much either way.

File: gnome/libgtop/Portfile [hunk 2]
```diff
@@ -42,4 +46,11 @@ configure.args      --x-includes=${prefix}/include \
                     --x-libraries=${prefix}/lib \
                     --disable-silent-rules
 
+# main.c: error: incompatible pointer to integer conversion passing 'char[1024]'
+# to parameter of type 'pid_t' (aka 'int') [-Wint-conversion]
+if {${os.platform} eq "darwin" && ${os.major} > 22} {
+    configure.cppflags-append \
+                    -Wno-error=int-conversion
+}
+
```
Comment (pmetzger) [inline gnome/libgtop/Portfile:55]: You should submit patches upstream later perhaps to fix this properly.

File: gnome/libgtop/Portfile (hunk 1 above)
Comment (barracuda156) [reply gnome/libgtop/Portfile:40]: I could have dropped this, of course.

**Pull request merged**